openerp/addons/base/res/res_currency.py

```diff
@@ -59,6 +59,15 @@ def _get_current_rate(self, cr, uid, ids, raise_on_no_rate=True, context=None):
                 raise osv.except_osv(_('Error!'),_("No currency rate associated for currency '%s' for the given period" % (currency.name)))
         return res
 
+    def _decimal_places(self, cr, uid, ids, name, arg, context=None):
```

Comment:
This method should be used in `ir.qweb` `MonetaryConverter`, which will be greatly simplified. The converter also needs to use the `currency_field` if the field is a monetary field, otherwise fallback to the current behavior.